plugin/index.js

```diff
@@ -1,43 +1,93 @@
+  return {
+    ...node,
+    leadingComments: [...leadingComments, {
+      type: 'CommentBlock',
+      value: ` babel-plugin-inline-import '${importedPath}' `
+    }]
+  }
+}
+
 export default function({ types: t }) {
-  class BabelInlineImport {
-    constructor() {
-      return {
-        visitor: {
-          ImportDeclaration: {
-            exit(path, state) {
-              const givenPath = path.node.source.value;
-              let reference = state && state.file && state.file.opts.filename;
-              const extensions = state && state.opts && state.opts.extensions;
-
-              if (BabelInlineImportHelper.shouldBeInlined(givenPath, extensions)) {
-                if (path.node.specifiers.length > 1) {
-                  throw new Error(`Destructuring inlined import is not allowed. Check the import statement for '${givenPath}'`);
-                }
-
-                const id = path.node.specifiers[0].local.name;
-                const content = BabelInlineImportHelper.getContents(givenPath, reference);
-                const variable = t.variableDeclarator(t.identifier(id), t.stringLiteral(content));
-
-                path.replaceWith({
-                  type: 'VariableDeclaration',
-                  kind: 'const',
-                  declarations: [variable],
-                  leadingComments: [
-                    {
-                      type: 'CommentBlock',
-                      value: ` babel-plugin-inline-import '${givenPath}' `
-                    }
-                  ]
-                });
-              }
-            }
+  return {
+    visitor: {
+      ImportDeclaration: {
+        exit(path, state) {
+          // get imported path
+          const importedPath = path.node.source.value;
+
+          // check if this is a path that needs to be imported
+          if (!shouldImportPathBeInlined(importedPath, state)) return;
+
+          // destructuring is unsupported
+          if (path.node.specifiers.length > 1) {
+            throw new Error(`Destructuring inlined import is not allowed. Check the import statement for '${givenPath}'`);
           }
+
+          // get file contents
+          const currentFilename = getCurrentFilename(state);
+          const content = BabelInlineImportHelper.getContents(importedPath, currentFilename);
+
+          // replace import with variable with file contents
+          path.replaceWith(annotate(importedPath, t.variableDeclaration('const', [
+            t.variableDeclarator(
+              t.identifier(path.node.specifiers[0].local.name),
+              t.stringLiteral(content)
+            )
+          ])));
         }
-      };
-    }
-  }
+      },
+
+      // support for require() expressions
+      Identifier(path, state) {
+        // only require
+        if (path.node.name !== 'require') return;
+
+        // only require global
+        const binding = path.scope.getBinding('require');
+        if (!binding) return;
```

Comment:
i think this is too strict

failure case:
```js
const foo = __importDefault(require("./bar.graphql"));
```

Reply:
Strict? It would apply in that case too. Do you mean not strict enough? Or do you mean I should also capture `__importDefault`?

Reply:
ah, my bad should've been more clear:
i tested your PR on a file
the file contained `const foo = __importDefault(require("./bar.graphql"));`
it failed on that transform because `!path.scope.getBinding('require')  === true`

Reply:
Right, because presumably this is the `__importDefault` from TypeScript which is expecting a module object which goes against what this plugin wants to provide (a string).

Either:

- You're manually calling `__importDefault` in which case you should just remove that code
- You're passing TypeScript compiled output into Babel with this plugin and it's rightfully confusing this Babel plugin.

There's not a lot I can do in this plugin otherwise -- The only thing I could do is explicitly banned `require()`'s that are wrapped with `__importDefault()` which is just a one-off fix for your setup.

Reply:
@mattkrick what do you think we should do regarding this issue?